src/Microsoft.AspNet.Mvc.Core/HttpResponseStreamWriter.cs

```diff
@@ -0,0 +1,179 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Framework.Internal;
+
+namespace Microsoft.AspNet.Mvc
+{
+    /// <summary>
+    /// Writes to the supplied <see cref="Stream"/> using the supplied <see cref="Encoding"/>.
+    /// It does not write the BOM and also does not close the stream.
+    /// </summary>
+    public class HttpResponseStreamWriter : TextWriter
+    {
+        private const int DefaultBufferSize = 1024;
+        private readonly Stream _stream;
+        private readonly int _bufferSize;
+        private byte[] _buffer;
+        private int _numOfBytesWrittenToBuffer;
+
+        public HttpResponseStreamWriter(Stream stream, Encoding encoding)
+            : this(stream, encoding, DefaultBufferSize)
+        {
+        }
+
+        public HttpResponseStreamWriter([NotNull] Stream stream, [NotNull] Encoding encoding, int bufferSize)
+        {
+            _stream = stream;
+            Encoding = encoding;
+            _bufferSize = bufferSize;
+        }
+
+        public override Encoding Encoding { get; }
+
+        public override void Write(char value)
+        {
+            var bytes = Encoding.GetBytes(new[] { value });
```

Comment:
This is going to cause issues. Take a look at how streamwriter does (using a buffer and the `Encoder` class).